Apply style to WMS Layer (#188)
Esri/arcgis-maps-sdk-flutter-samples: lib/samples/apply_style_to_wms_layer/README.md

# Apply style to WMS layer

Change the style of a Web Map Service (WMS) layer.

![Image of apply style to WMS layer](apply_style_to_wms_layer.png)

## Use case

Layers hosted on WMS may have different pre-set styles available to apply to them. Swapping between these styles can help during visual examination of the data. For example, increasing the contrast of satellite images can help in identifying urban and agricultural areas within forested areas.

## How to use the sample

Once the layer loads, toggle between the first and second styles of the WMS layer using the dropdown menu.

## How it works

1. Create a `WmsLayer` specifying the URL of the service and the layer names you want `WmsLayer.withUriAndLayerNames(uri: Uri.parse(url), layerNames: names)`.

> 2. When the layer is done loading, get its list of style strings using `wmsLayer.layerInfos.styles`.

Reviewer: ```suggestion
2. When the layer is done loading, get its list of style strings using `WmsLayer.layerInfos.styles`.
```

# Apply style to WMS layer

Change the style of a Web Map Service (WMS) layer.

![Image of apply style to WMS layer](apply_style_to_wms_layer.png)

## Use case

Layers hosted on WMS may have different pre-set styles available to apply to them. Swapping between these styles can help during visual examination of the data. For example, increasing the contrast of satellite images can help in identifying urban and agricultural areas within forested areas.

## How to use the sample

> Once the layer loads, the toggle button will be enabled. Click it to toggle between the first and second styles of the WMS layer.

Reviewer: ```suggestion
Once the layer loads, toggle between the first and second styles of the WMS layer using the dropdown menu.
```
Or words to that effect that better describe our UI.

## How it works

1. Create a `WmsLayer` specifying the URL of the service and the layer names you want `WmsLayer.withUriAndLayerNames(uri: Uri.parse(url), layerNames: names)`.
2. When the layer is done loading, get one of its sublayer `wmsLayer.sublayers.first as WmsSublayer`.
3. After obtaining one of its sub layers, get its list of style strings using `firstSublayer.sublayerInfo.styles`.

> 4. Set one of the styles using `firstSublayer.currentStyle = styles[0]`.

Reviewer: `wmsLayer.sublayers.first as WmsSublayer`. This cast had me down a rabbit hole of API ref! It looks like the sample design is based on another implementation - there seems to be some minor difference between the sdks here with whether sublayers themselves actually come back from that sublayers property. It looks like our API is correct (per the API design) but I think this is what you want to do in this scenario for our API:

When the layer is done loading, get its list of style strings using `wmsLayer.layerInfos.first.styles`.

Then ideally we won't display a cast in the readme steps so maybe the next step could end up being something like:
Set one of the styles to the first sublayer using `WmsSublayer.currentStyle = style.first`.

Author: I think it's missing `.sublayerInfo` before `.styles` in :  `wmsLayer.layerInfos.first.styles`.
